scrapper.py

```diff
@@ -25,39 +39,159 @@ class Crawler:
+            response = requests.get(url)
+            sleep_period = random.randint(1, 3)
+            sleep(sleep_period)
+
+            if not response.ok:
+                continue
+
+            soup = BeautifulSoup(response.text, 'lxml')
+            self._extract_url(soup)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        title = article_bs.find('h1', class_='page_title').text.strip()
+        back_to_seed = article_bs.select_one('nav ol li:nth-child(3) a')['href']
+        seed_bs = BeautifulSoup(requests.get(back_to_seed).text, 'lxml')
+        sections = seed_bs.find_all("div", class_="obj_article_summary")
+        for section in sections:
+            if title in section.text:
+                found_article_url_bs = section.find('a', class_='obj_galley_link pdf')
+                if found_article_url_bs:
+                    art_soup = BeautifulSoup(requests.get(found_article_url_bs['href']).text, 'lxml')
+                    download_pdf = art_soup.find('a', class_='download')['href']
+                    pdf = PDFRawFile(download_pdf, self.article_id)
+                    pdf.download()
+                    self.article.text = pdf.get_text().split('СПИСОК ЛИТЕРАТУРЫ')[0]
+
+    def _fill_article_with_meta_information(self, article_bs):
+        # title
+        try:
+            self.article.title = article_bs.find('h1', class_='page_title').text.strip()
+        except AttributeError:
+            self.article.title = 'NOT FOUND'
+        # author
+        try:
+            self.article.author = article_bs.find('ul', class_='item authors').find('li').find('span').text.strip()
+        except AttributeError:
+            self.article.title = 'NOT FOUND'
+
+        # topics
+        try:
+            self.article.topics = article_bs.find('div', class_='item keywords').find(
+                'span', class_='value').text.strip().replace('\t', "").split(', ')
+        except AttributeError:
+            self.article.title = 'NOT FOUND'
+
+        # date
+        try:
+            date = article_bs.find('div', class_='item published').find('div', class_='value').text.strip()
+            self.article.date = datetime.strptime(date, '%Y-%m-%d')
+        except AttributeError:
+            self.article.date = datetime.strptime('2021', '%Y')
+
+    def parse(self):
+        response = requests.get(url=self.article_url)
+
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        if self.article.text:
+            self._fill_article_with_meta_information(article_bs)
+            return self.article
```

Comment:
this is not needed, you return anyway in next line

Reply:
Thank you, I've removed it